### PR TITLE
add documentation on parted partition alignment

### PR DESCRIPTION
**Pull request by wfeldt**

### CODE REVIEW

File: doc/parted.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+## Normal Disk
+
+- parted aligns to 1MiB
+
+## RAID
+
+### setup using mdadm
+
+	mdadm --create /dev/md0 -R -l 0 -c 32 -n 3 /dev/sdb /dev/sdc /dev/sdd
```
Comment (kobliha) [inline doc/parted.md]: Please use long format of parameters, this is explanatory documentation

```
-R -> --run
-l -> --level=0
-c -> --chunk=32
-n -> --raid-devices=?
```

File: doc/parted.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# parted And Partition Alignment
+
+## General
+
+- parted will try to align newly created partitions. But only if the partition
+start/end are a bit 'fuzzy'. For example:
+
+ - 'parted p ext2 1% 2%' will give nicely aligned partitions; so does
+ - 'parted p ext2 1MB 2MB'. But the supposedly more correct
+ - 'parted p ext2 1MiB 2MiB' will **not** do what you (well, at least I did) expect. The partition size will be 1 MiB + 1 sector.
```
Comment (aschnell) [inline doc/parted.md]: The behaviour is indeed surprising and not mentioned in the man-page but only the info page.

File: doc/parted.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ 
+- parted does **not** align to cylinder sizes
```
Comment (aschnell) [inline doc/parted.md]: You should mention the --align option of parted.

File: doc/parted.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+- parted will align to 1MiB, unless your the underlying raid config does
+have a chunk size which is not a power of 2 smaller than 1MiB. In that case parted aligns to
+chunk size * 'usable disks' - see details below.
```
Comment (aschnell) [inline doc/parted.md]: You should mention that parted uses topology information provided by the kernel or libblkid (see http://git.savannah.gnu.org/cgit/parted.git/tree/libparted/arch/linux.c#n3206). The alignment_offset and optimal_io_size are also available under /sys.

**Comment by kobliha** [comment]
@wfeldt @aschnell JFYI, I have asked Petr Uzel for comments, so, please, wait for them before merging

File: doc/parted.md (hunk 2 above)
Comment (knot) [inline doc/parted.md]: Yes, this should be noted in the manpage. {M,G,T}iB addresses are taken as exact. If it is problematic (except missing documentation), we might discuss how to improve it.

File: doc/parted.md [hunk 5]
```diff
@@ -61,11 +66,45 @@ _This seems to be inconsistent to me. 1MiB * 'usable disks' would be more logica
...
+
+For parted's strategy look at `libparted/arch/linux.c::linux_get_optimum_alignment()`.
+
 
 ## Conclusion
 
 Letting parted decide on the exact partition sizes seems to be prone to
 achive unpredictable results. It might be better to deal with it inside
-libstorage.
+libstorage. As shown, aligning partitions is no rocket science. :-)
```
Comment (aschnell) [inline doc/parted.md]: Yeap, no magic.

**Comment by aschnell** [comment]
LGTM

**Pull request merged**